Mathlib/Data/Nat/Parity.lean

```diff
@@ -236,6 +236,12 @@ theorem one_add_div_two_mul_two_of_odd (h : Odd n) : 1 + n / 2 * 2 = n := by
   rw [← odd_iff.mp h, mod_add_div']
 #align nat.one_add_div_two_mul_two_of_odd Nat.one_add_div_two_mul_two_of_odd
 
+theorem succ_mod_two_eq_zero_iff {m : ℕ} : (m + 1) % 2 = 0 ↔ m % 2 = 1 := by
+  simp [← Nat.even_iff, ← Nat.not_even_iff, parity_simps]
+
+theorem succ_mod_two_eq_one_iff {m : ℕ} : (m + 1) % 2 = 1 ↔ m % 2 = 0 := by
+  simp [← Nat.even_iff, ← Nat.not_even_iff, parity_simps]
+
```

Comment:
Let's move these to right after `even_add_one`, which is the earliest they can go.